core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/chunked.kt

```diff
@@ -7,12 +7,28 @@ import org.jetbrains.kotlinx.dataframe.columns.ColumnGroup
 import org.jetbrains.kotlinx.dataframe.columns.FrameColumn
 import org.jetbrains.kotlinx.dataframe.columns.ValueColumn
 import org.jetbrains.kotlinx.dataframe.impl.getListType
+import org.jetbrains.kotlinx.dataframe.impl.splitByIndices
 import org.jetbrains.kotlinx.dataframe.nrow
 import org.jetbrains.kotlinx.dataframe.type
 
+/**
+ * Creates a [FrameColumn] from [this] by splitting the dataframe into
+ * smaller ones, based on the given [startIndices].
+ */
+public fun <T> DataFrame<T>.chunked(startIndices: Iterable<Int>, name: String = "groups"): FrameColumn<T> =
```

Comment:
I always saw DataColumn.createFrameColumn(Iterable<Int>) was mostly an internal API, i wouldn't expose it like this because it's hard to imagine a use case for users

Reply:
I can make it `chunkedImpl` maybe :)